exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/flatten/FlattenRecordBatch.java

```diff
@@ -267,10 +276,21 @@ private TransferPair getFlattenFieldTransferPair(FieldReference reference) {
     if (flattenField instanceof RepeatedMapVector) {
       tp = ((RepeatedMapVector)flattenField).getTransferPairToSingleMap(reference.getAsNamePart().getName(), oContext.getAllocator());
     } else {
-      final ValueVector vvIn = RepeatedValueVector.class.cast(flattenField).getDataVector();
-      // vvIn may be null because of fast schema return for repeated list vectors
-      if (vvIn != null) {
-        tp = vvIn.getTransferPair(reference.getAsNamePart().getName(), oContext.getAllocator());
+      try {
+        final ValueVector vvIn = RepeatedValueVector.class.cast(flattenField).getDataVector();
+        // vvIn may be null because of fast schema return for repeated list vectors
+        if (vvIn != null) {
+          tp = vvIn.getTransferPair(reference.getAsNamePart().getName(), oContext.getAllocator());
+        }
+      } catch(ClassCastException ex) {
```

Comment:
We can apply similar changes above here ( catch ClassCastException -> instancOf check), also, logger same message.